### PR TITLE
1290/use dayjs instead of moment

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #1290 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This swaps out Moment date library in favor of Day.js across all four packages of the monorepo.

## How Can This Be Tested/Reviewed?

Just generally try out some of the obvious code paths that deal with dates…view a listing, submit an application, update a listing.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/listings/Aside.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React, { useContext, useMemo, useCallback } from "react"
```
Comment (dominikx96) [inline sites/partners/src/listings/Aside.tsx:1]: I see you get rid of the `saveAndExit ` function, I think we can also remove `useCallback` and a couple of another unused elements in this file.

**Review by dominikx96 — APPROVED** [review]
LGTM

File: sites/partners/src/listings/PaperListingForm/formTypes.ts [hunk 2]
```diff
@@ -92,7 +92,7 @@ export const formDefaults: FormListing = {
   applicationDueTime: null,
   applicationFee: null,
   applicationMethods: [],
-  applicationOpenDate: new Date(moment().subtract(10).format()),
+  applicationOpenDate: new Date(dayjs().subtract(10).format()),
```
Comment (jaredcwhite) [inline sites/partners/src/listings/PaperListingForm/formTypes.ts:95]: This was already an issue, but I think this code is missing the second argument so we're not sure what the unit should be. Subtract 10 seconds? Minutes? Days?

**Comment by jaredcwhite** [comment]
My above question aside, this should be good to merge.

File: sites/partners/src/listings/PaperListingForm/formTypes.ts (hunk 2 above)
Comment (seanmalbert) [reply sites/partners/src/listings/PaperListingForm/formTypes.ts:95]: I don't think we need this. If it's "now" that will still be in the past once it's saved.

File: ui-components/__tests__/page_components/LeasingAgent.test.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { render, cleanup } from "@testing-library/react"
 import { LeasingAgent } from "../../src/page_components/listing/listing_sidebar/LeasingAgent"
 import { ArcherListing } from "@bloom-housing/backend-core/types/src/archer-listing"
 import { Listing } from "@bloom-housing/backend-core/types"
-import moment from "moment"
+import dayjs from "dayjs"
```
Comment (emilyjablonski) [inline ui-components/__tests__/page_components/LeasingAgent.test.tsx:6]: Looks like we're not using this import

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Some clean up above

File: backend/core/package.json [hunk 4]
```diff
@@ -66,7 +66,6 @@
     "ioredis": "^4.24.4",
     "joi": "^17.3.0",
     "jwt-simple": "^0.5.6",
-    "moment": "^2.29.1",
```
[the remark this answers is not in the document]
Comment (jaredcwhite) [reply backend/core/package.json:69]: I did add it to the file, up just below cloudinary.

File: sites/partners/src/listings/Aside.tsx [hunk 5]
```diff
@@ -38,7 +38,7 @@ const Aside = ({
   const recordUpdated = useMemo(() => {
     if (!listing) return null
 
-    const momentDate = moment(listing.updatedAt)
```
[the remark this answers is not in the document]
Comment (jaredcwhite) [reply sites/partners/src/listings/Aside.tsx:41]: Yeah, I'd like to tackle all that stuff as part of overall date refactoring so we can reduce the number of one-off formatting statements.

**Pull request merged**